TestRateLimitConfigReload is unstable

## Flaky Test

### Which jobs are failing
```
2024-04-11T18:31:13.2054834Z --- FAIL: TestRateLimitConfigReload (12.08s)
2024-04-11T18:31:13.2055423Z     config_test.go:83: 
2024-04-11T18:31:13.2056189Z         	Error Trace:	/home/runner/work/pd/pd/tests/server/config/config_test.go:83
2024-04-11T18:31:13.2057163Z         	Error:      	Expected value not to be nil.
2024-04-11T18:31:13.2057806Z         	Test:       	TestRateLimitConfigReload
```

### CI link
https://github.com/tikv/pd/actions/runs/8651545822/job/23722461062

### Reason for failure (if possible)


### Anything else

--- a/tests/server/config/config_test.go
+++ b/tests/server/config/config_test.go
@@ -32,7 +32,6 @@ import (
 	tu "github.com/tikv/pd/pkg/utils/testutil"
 	"github.com/tikv/pd/pkg/utils/typeutil"
 	"github.com/tikv/pd/pkg/versioninfo"
-	"github.com/tikv/pd/server"
 	"github.com/tikv/pd/server/config"
 	"github.com/tikv/pd/tests"
 )
@@ -74,11 +73,7 @@ func TestRateLimitConfigReload(t *testing.T) {
 
 	oldLeaderName := leader.GetServer().Name()
 	leader.GetServer().GetMember().ResignEtcdLeader(leader.GetServer().Context(), oldLeaderName, "")
-	var servers []*server.Server
-	for _, s := range cluster.GetServers() {
-		servers = append(servers, s.GetServer())
-	}
-	server.MustWaitLeader(re, servers)
+	re.NotEmpty(cluster.WaitLeader())
 	leader = cluster.GetLeaderServer()
 	re.NotNil(leader)
 	re.True(leader.GetServer().GetServiceMiddlewarePersistOptions().IsRateLimitEnabled())

